Mathlib/SetTheory/Surreal/Basic.lean

```diff
@@ -439,6 +427,643 @@ lemma bddBelow_of_small (s : Set Surreal.{u}) [Small.{u} s] : BddBelow s := by
 
 end Surreal
 
+/-!
+### Surreal multiplication
+
+This section carries out the main inductive argument that proves the following three main theorems:
+* P1: being numeric is closed under multiplication,
+* P2: multiplying a numeric pregame by equivalent numeric pregames results in equivalent pregames,
+* P3: the product of two positive numeric pregames is positive (`mul_pos`).
+
+This is Theorem 8 in [Conway2001], or Theorem 3.8 in [SchleicherStoll]. P1 allows us to define
+multiplication as an operation on numeric pregames, P2 says that this is well-defined as an
+operation on the quotient by `PGame.Equiv`, namely the surreal numbers, and P3 is an axiom that
+needs to be satisfied for the surreals to be a `OrderedRing`.
+
+We follow the proof in [schleicher_stoll], except that we use the well-foundedness of
+the hydra relation `CutExpand` on `Multiset PGame` instead of the argument based
+on a depth function in the paper.
+
+In the argument, P3 is stated with four variables `x₁`, `x₂`, `y₁`, `y₂` satisfying `x₁ < x₂` and
+`y₁ < y₂`, and says that `x₁ * y₂ + x₂ * x₁ < x₁ * y₁ + x₂ * y₂`, which is equivalent to
+`0 < x₂ - x₁ → 0 < y₂ - y₁ → 0 < (x₂ - x₁) * (y₂ - y₁)`, i.e.
+`@mul_pos PGame _ (x₂ - x₁) (y₂ - y₁)`. It has to be stated in this form and not in terms of
+`mul_pos` because we need to show show P1, P2 and (a specialized form of) P3 simultaneously, and
+for example `P1 x y` will be deduced from P3 with variables taking values simpler than `x` or `y`
+(among other induction hypotheses), but if you subtract two pregames simpler than `x` or `y`,
+the result may no longer be simpler.
+
+The specialized version of P3 is called P4, which takes only three arguments `x₁`, `x₂`, `y` and
+requires that `y₂ = y` or `-y` and that `y₁` is a left option of `y₂`. After P1, P2 and P4 are
+shown, a further inductive argument (this time using the `GameAdd` relation) proves P3 in full.
+
+Our proof features a clear separation into components/submodules:
+* calculation (e.g. ...),
+* specialize induction hypothesis to a form easier to apply
+  (that direct takes in `IsOption` arguments),
+* application of specialized indution hypothesis,
+* symmetry verification,
+* verification of `CutExpand` relations,
+* `Numeric`ity of options (filled in at the last moment ).
+and we utilize symmetry (permutation and negation of arguments) to minimize calculation.
+
+  strategy: extract specialized versions of the induction hypothesis that easier to apply
+  (example: `ih1`, ...),
+  show they are invariant under certain symmetries (permutation and negation of variables),
+  and that the induction hypothesis indeed implies the specialized versions.
+  (Actually the induction hypotheses themselves already have those symmetries ..? No, not usually
+  the negation symmetries).
+
+  add the numeric hypothesis only at the last moment ..
```

Comment:
This part is in draft form and that was the main reason why I didn't open a PR. Not sure how much I could recall about what I intended to say ...